app/templates/new-signup.hbs

```diff
@@ -0,0 +1,46 @@
+{{page-title "New Sign Up"}}
+
+{{#if (eq this.currentStep this.FIRST_STEP)}}
+  <NewSignup::Info
+    @onClick={{this.firstStep}}
+    @currentStep={{this.currentStep}}
+  />
+{{/if}}
+
+{{#if (eq this.currentStep this.SECOND_STEP)}}
+  <NewSignup::Input
+    @onClick={{this.secondStep}}
+    @currentStep={{this.currentStep}}
+    @onChange={{this.handleInputChange}}
+    @isButtonDisabled={{this.isButtonDisabled}}
+    @isLoading={{this.isLoading}}
+  />
+{{/if}}
+
+{{#if (eq this.currentStep this.THIRD_STEP)}}
+  <NewSignup::Input
+    @onClick={{this.thirdStep}}
+    @currentStep={{this.currentStep}}
+    @onChange={{this.handleInputChange}}
+    @isButtonDisabled={{this.isButtonDisabled}}
+    @isLoading={{this.isLoading}}  
+  />
+{{/if}}
+
+{{#if (eq this.currentStep this.FOURTH_STEP)}}
+  <NewSignup::Input
+    @onClick={{this.fourthStep}}
+    @currentStep={{this.currentStep}}
+    @onChange={{this.handleInputChange}}
+    @isButtonDisabled={{this.isButtonDisabled}}
+    @isLoading={{this.isLoading}}
+    @error={{this.error}}
+  />
+{{/if}}
+
+{{#if (eq this.currentStep this.LAST_STEP)}}
+  <NewSignup::Info
+    @onClick={{this.lastStep}}
+    @currentStep={{this.currentStep}}
+  />
+{{/if}}
```

Comment:
What do you think about simplifying this mapping in an array or array of objects as the code seems very repetitive

Reply:
I was doing that only, ankush suggested to it this way

Reply:
ok got it, maybe that would be over-simplifying the code and eventually lead to an increase in complexity.